x-pack/plugins/security/server/lib/saved_objects_client/secure_saved_objects_client_wrapper.test.ts

```diff
@@ -0,0 +1,1341 @@
+    generalError,
+    decorateGeneralError: jest.fn().mockReturnValue(generalError),
+  };
+};
+
+const createMockAuditLogger = () => {
+  return {
+    savedObjectsAuthorizationFailure: jest.fn(),
+    savedObjectsAuthorizationSuccess: jest.fn(),
+  };
+};
+
+const createMockActions = () => {
+  return {
+    savedObject: {
+      get(type: string, action: string) {
+        return `mock-saved_object:${type}/${action}`;
+      },
+    },
+  } as Actions;
+};
+
+const createCheckSavedObjectsPrivileges = (checkPrivilegesImpl: () => Promise<any>) => {
+  const mockAuditLogger = createMockAuditLogger();
+  const mockErrors = createMockErrors();
+  const mockActions = createMockActions();
+
+  const checkSavedObjectsPrivileges = jest.fn(
+    checkSavedObjectsPrivilegesFactory({
+      actionsService: mockActions,
+      auditLogger: mockAuditLogger,
+      checkPrivilegesWithRequest: () => ({
+        atSpace: jest.fn().mockImplementationOnce(checkPrivilegesImpl),
+        atSpaces: jest.fn().mockRejectedValue('atSpaces should not be called'),
+        globally: jest.fn().mockRejectedValue('globally should not be called'),
+      }),
+      errors: mockErrors,
+      request: null as any,
+      spacesEnabled: true,
+    })
+  );
+
+  return {
+    mockActions,
+    mockAuditLogger,
+    mockErrors,
+    checkSavedObjectsPrivileges,
+  };
+};
+
+describe('#errors', () => {
+  test(`assigns errors from constructor to .errors`, () => {
+    const errors = Symbol();
+
+    const { checkSavedObjectsPrivileges } = createCheckSavedObjectsPrivileges(() => {
+      throw new Error('An actual error would happen here');
+    });
+
+    const client = new SecureSavedObjectsClientWrapper(({
+      checkSavedObjectsPrivileges,
+      errors,
+    } as unknown) as SecureSavedObjectsClientWrapperDeps);
+
+    expect(client.errors).toBe(errors);
+  });
+});
+
+describe(`spaces disabled`, () => {
+  describe('#create', () => {
+    test(`throws decorated GeneralError when checkPrivileges.globally rejects promise`, async () => {
+      const type = 'foo';
+      const {
+        mockErrors,
+        mockAuditLogger,
+        checkSavedObjectsPrivileges,
+      } = createCheckSavedObjectsPrivileges(() => {
+        throw new Error('An actual error would happen here');
+      });
+
+      const client = new SecureSavedObjectsClientWrapper({
+        baseClient: (null as unknown) as SavedObjectsClientContract,
+        checkSavedObjectsPrivileges,
+        errors: mockErrors,
+      });
+
+      await expect(client.create(type, {})).rejects.toThrowError(mockErrors.generalError);
+      expect(checkSavedObjectsPrivileges).toHaveBeenCalledWith(type, 'create', undefined, {
+        attributes: {},
+        options: {},
+        type,
+      });
+      expect(mockErrors.decorateGeneralError).toHaveBeenCalledTimes(1);
+      expect(mockAuditLogger.savedObjectsAuthorizationFailure).not.toHaveBeenCalled();
```

Comment:
I decided (for now) to keep the audit logging assertions as part of this test suite. We're technically testing an implementation detail of a dependency now, but I feel like it's important to verify that it works from the service level, and the Saved Object Client is as close to an API as we can get without mocking out even more services.

We have audit logging tests as part of https://github.com/elastic/kibana/blob/764322cc7f3a6f3b082a0044af151e1b2a1da21f/x-pack/plugins/security/server/lib/authorization/saved_objects/check_saved_objects_privileges.test.ts, so I don't have a strong opposition to removing the checks here if we want to simplify this test suite a bit.